core/modules/widgets/checkbox.js

```diff
@@ -277,14 +296,19 @@ Selectively refreshes the widget if needed. Returns true if the widget or any of
 */
 CheckboxWidget.prototype.refresh = function(changedTiddlers) {
 	var changedAttributes = this.computeAttributes();
-	if(changedAttributes.tiddler || changedAttributes.tag || changedAttributes.invertTag || changedAttributes.field || changedAttributes.index || changedAttributes.listField || changedAttributes.filter || changedAttributes.checked || changedAttributes.unchecked || changedAttributes["default"] || changedAttributes["class"] || changedAttributes.disabled) {
+	if(changedAttributes.tiddler || changedAttributes.tag || changedAttributes.invertTag || changedAttributes.field || changedAttributes.index || changedAttributes.listField || changedAttributes.listIndex || changedAttributes.filter || changedAttributes.checked || changedAttributes.unchecked || changedAttributes["default"] || changedAttributes.indeterminate || changedAttributes["class"] || changedAttributes.disabled) {
```

Comment:
BTW, this line contains a minor bugfix for #6561. I added the `listIndex` field in #6561, but forgot to include it in this list of changed attributes. So if a user were to make `listIndex` an attribute that comes from a text reference (for some use case I can't think of), then a change in the value referred to by that text reference would not cause the checkbox to re-render. Would be very rare to occur, but I noticed it while I was making this PR so I might as well fix it.

Reply:
Are there any attributes that do not warrant a refresh? If not, we can use the following check:

`if($tw.utils.count(changedAttributes) > 0)`